spacy/pipeline/spancat.py

```diff
@@ -272,6 +272,24 @@ def predict(self, docs: Iterable[Doc]):
         scores = self.model.predict((docs, indices))  # type: ignore
         return indices, scores
 
+    def suggest_candidates(
+        self, docs: Iterable[Doc], spankey: str = "candidates"
+    ) -> List[Doc]:
+        """Use the spancat suggester to add a list of span candidates to a list of docs
+
+        docs (Iterable[Doc]): The documents to predict.
+        RETURNS: A list of Docs.
+
+        DOCS: WIP
+        """
+        suggester_output = self.suggester(docs, ops=self.model.ops)
+
+        for candidates, doc in zip(suggester_output, docs):
+            doc.spans[spankey] = []
+            for index in candidates.dataXd:
+                doc.spans[spankey].append(doc[index[0] : index[1]])
+        return docs
```

Comment:
`mypy` complains here because you're not returning a `List` as the typing indicates. You could either retype this to return `Iterable[Doc]` which is the correct type of `docs`, or you could return `list(docs)` instead.

Reply:
Do you need to return the docs? `set_annotations` doesn't return the docs.

Reply:
You're right, we might as well just not return the docs, given that it's all done in-place. That also resolves that ambiguity for the user.